client/client.go

```diff
@@ -34,6 +34,53 @@ import (
 	"github.com/canonical/pebble/internals/wsutil"
 )
 
+type Requester interface {
+	// Do performs the HTTP transaction using the provided options.
+	Do(ctx context.Context, opts *RequestOptions) (*RequestResponse, error)
+
+	// Transport returns the HTTP transport in use by the underlying HTTP client.
+	Transport() *http.Transport
+}
+
+type RequestType int
+
+const (
+	SyncRequest RequestType = iota
```

Comment:
I was leaning toward the choice that would result in making the most common case as compact as possible (not having to set Type).
```
flotter@pc:~/pebble/client$ grep -r doAsync | wc -l
4
flotter@pc:~/pebble/client$ grep -r doSync | wc -l
19
```
But, I do still prefer setting the Type just to make the code more explicit, and since we have wrappers ```doSync``` and ```doAsync```, the cost is negligible. I changed it so raw comes first.